src/main.rs

```diff
@@ -40,22 +55,42 @@ enum BackendArgs {
     WASM,
 }
 
+#[derive(Clone, Copy, Debug, ValueEnum)]
+enum EmitArgs {
+    None,
+    Ir,
+    C,
+}
+
 lalrpop_mod!(pub rascal_grammar);
 
 fn main() {
     let args = Args::parse();
+    let save_c = args.emit.iter().any(|x| matches!(x, EmitArgs::C));
+    let save_ir = args.emit.iter().any(|x| matches!(x, EmitArgs::Ir));
     let src_file = fs::read_to_string(args.infile).expect("ERROR: couldn't find source file");
     let root = rascal_grammar::RootParser::new().parse(&src_file).unwrap();
     // Perform semantic checks and type checking
     let mut state = semantic::new_state(root);
     state.build().unwrap();
+    if save_ir {
+        let serialized_ir = serde_json::to_string(&state.build_stack).expect("Serialization error");
+        let mut file =
+            File::create(ProgramState::IR_OUTPUT_FILENAME).expect("Cannot create IR file");
+        write!(&mut file, "{serialized_ir}").expect("Cannot write to IR file");
+    } else {
+        fs::remove_file(ProgramState::IR_OUTPUT_FILENAME).expect("Unable to delete IR output file");
+    }
```

Comment:
I don't think this block does what you intended. If you omit the `--emit` flag altogether (or just don't emit the `ir`) you get a panic as you're trying to delete an IR file that doesn't exist.

Reply:
The default behavior if you don't provide anything to `--emit` should just be the final binary. So either a linked executable or a `.wasm`. I think you could probably just remove the `else` statement here to get this.

Reply:
Having to temporarily save the `.c` file then delete it if we _don't_ want to save makes sense though since that is the easiest way to pass it to GCC.